pkg/pillar/types/sriovtypes.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	NicLinuxPath     = "/sys/class/net/"
+	NumvfsDevicePath = "/device/sriov_numvfs"
+	VfCountFieldName = "sriov-vf-count"
+)
+
+// Must match fields of EthVF in devcommon.proto
+type EthVF struct {
+	Index        uint8
+	PciLong      string // BFD notation
+	Iface        string
+	UsedByUUID   uuid.UUID
+	Mac          net.HardwareAddr
+	VlanId       uint16
+	Logicallabel string
+}
+
+func (pf *EthVF) fromAdapter(rhs EthVF) {
+	if rhs.Mac != nil {
+		pf.Mac = rhs.Mac
+	}
+	if rhs.Index != 0 {
+		pf.Index = rhs.Index
+	}
+	if rhs.VlanId != 0 {
+		pf.VlanId = rhs.VlanId
+	}
+}
+
+func (vf *EthVF) setupLink() error {
+	pf, err := netlink.LinkByName(vf.Iface)
+	if err != nil {
+		return fmt.Errorf("Failed to find physical function %s: %v", vf.Iface, err)
+	}
+	if err = netlink.LinkSetVfHardwareAddr(pf, int(vf.Index), vf.Mac); err != nil {
+		return fmt.Errorf("Failed to set vf %d mac address: %v", vf.Index, err)
+	}
+
+	return nil
+}
+
+func (vf *EthVF) setupVlan() error {
+	if vf.VlanId == 0 {
+		// Either vlan is not initialized or not used
+		return nil
+	}
+	pf, err := netlink.LinkByName(vf.Iface)
+	if err != nil {
+		return fmt.Errorf("Failed to find physical function %s: %v", vf.Iface, err)
+	}
+
+	if err = netlink.LinkSetVfVlan(pf, int(vf.Index), int(vf.VlanId)); err != nil {
+		return fmt.Errorf("Failed to set vf %d vlan: %v", vf.Index, err)
+	}
+	return nil
+}
+
+// VFList is list of VF for given PF (Eth device)
+type VFList struct {
+	Count uint8
+	Data  []EthVF
+}
+
+// UsedBy finds VF used by app UUID
+func (l *VFList) UsedBy(appUuid uuid.UUID) *EthVF {
+	for _, vf := range l.Data {
+		if vf.UsedByUUID == appUuid {
+			return &vf
+		}
+	}
+	return nil
+}
+
+// UsedById finds VF used by VF id
+func (l *VFList) UsedById(id uint8) *EthVF {
```

Comment:
No callers of this function?

Reply:
Since we changed approach, this one is not needed as well